src/Microsoft.NetCore.Analyzers/Core/Security/UseAutoValidateAntiforgeryToken.cs

```diff
@@ -0,0 +1,287 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Analyzer.Utilities;
+using Analyzer.Utilities.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.NetCore.Analyzers.Security
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class UseAutoValidateAntiforgeryToken : DiagnosticAnalyzer
+    {
+        internal const string DiagnosticId = "CA5391";
+        private static readonly LocalizableString s_Title = new LocalizableResourceString(
+            nameof(SystemSecurityCryptographyResources.UseAutoValidateAntiforgeryToken),
+            SystemSecurityCryptographyResources.ResourceManager,
+            typeof(SystemSecurityCryptographyResources));
+        private static readonly LocalizableString s_Message = new LocalizableResourceString(
+            nameof(SystemSecurityCryptographyResources.UseAutoValidateAntiforgeryTokenMessage),
+            SystemSecurityCryptographyResources.ResourceManager,
+            typeof(SystemSecurityCryptographyResources));
+        private static readonly LocalizableString s_Description = new LocalizableResourceString(
+            nameof(SystemSecurityCryptographyResources.UseAutoValidateAntiforgeryTokenDescription),
+            SystemSecurityCryptographyResources.ResourceManager,
+            typeof(SystemSecurityCryptographyResources));
+
+        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(
+                DiagnosticId,
+                s_Title,
+                s_Message,
+                DiagnosticCategory.Security,
+                DiagnosticHelpers.DefaultDiagnosticSeverity,
+                isEnabledByDefault: DiagnosticHelpers.EnabledByDefaultIfNotBuildingVSIX,
```

Comment:
>EnabledByDefaultIfNotBuildingVSIX [](start = 54, length = 33)

Given that we won't necessarily be able to see all of an application's binaries (might be in an unreferenced assembly) to see if there's a global antiforgery filter, I think this may end up being noisy.

Reply:
Oh, hrm, maybe if we're checking to see if there's at least one [ValidateAntiForgeryToken], then maybe it's okay.

---
In reply to: [299224935](https://github.com/dotnet/roslyn-analyzers/pull/2646#discussion_r299224935) [](ancestors = 299224935)

Reply:
But to check if a least one [ValidateAntiForgeryToken] is used, we still need to register operation on every class or method, it costs, like check global anti forgery filter.

Reply:
Right, I wasn't worried about the cost, but whether the rule should be disabled by default.